Gems/Terrain/Code/Source/Components/TerrainWorldDebuggerComponent.cpp

```diff
@@ -195,6 +269,99 @@ namespace Terrain
         debugDisplay.DrawWireBox(aabb.GetMin(), aabb.GetMax());
     }
 
+    void TerrainWorldDebuggerComponent::DrawQueries(
+        const AzFramework::ViewportInfo& viewportInfo, AzFramework::DebugDisplayRequests& debugDisplay)
+    {
+        AZ_PROFILE_FUNCTION(Terrain);
+
+        if (!m_configuration.m_debugQueries.m_drawQueries)
+        {
+            return;
+        }
+
+        // Early out if none of our draw toggles are enabled.
+        if (!(m_configuration.m_debugQueries.m_drawHeights || m_configuration.m_debugQueries.m_drawNormals))
+        {
+            return;
+        }
```

Comment:
Even better.. make this a function in `TerrainDebugQueryVisualizerConfig` since there are several there already.

Reply:
I put them on different lines because there's a precedence to the logic. If drawQueries is false but drawHeights is true, it should still early out. I can combine it, but it becomes harder to read IMO, especially since it's all negative logic:
```
if (!m_configuration.m_debugQueries.m_drawQueries || 
    !(m_configuration.m_debugQueries.m_drawHeights || m_configuration.m_debugQueries.m_drawNormals))
{
    return;
}
```
I'm inclined to leave this one as-is for readability.